### PR TITLE
 use the upstream RBAC roles for reconciliation

### PR DESCRIPTION
**Pull request by deads2k**
This starts using the upstream RBAC post starthook for reconciliation of RBAC resources.  We keep the duplication of the core upstream roles to provide webconsole annotations another release.

### CODE REVIEW

**Comment by enj** [comment]
> provide webconsole annotations another release

Which annotations is that?

File: pkg/cmd/server/bootstrappolicy/constants.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ const (
 	NodeAdminRoleBindingName          = NodeAdminRoleName + "s"
 	SDNReaderRoleBindingName          = SDNReaderRoleName + "s"
 	WebHooksRoleBindingName           = WebHooksRoleName + "s"
-	DiscoveryRoleBindingName          = DiscoveryRoleName + "-binding"
+	DiscoveryRoleBindingName          = DiscoveryRoleName
```
Comment (enj) [inline pkg/cmd/server/bootstrappolicy/constants.go]: Drop this const and switch the code to use `NewClusterBinding`

File: pkg/cmd/server/bootstrappolicy/constants.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ const (
 	SDNManagerRoleName        = "system:sdn-manager"
 	OAuthTokenDeleterRoleName = "system:oauth-token-deleter"
 	WebHooksRoleName          = "system:webhook"
-	DiscoveryRoleName         = "system:discovery"
+	DiscoveryRoleName         = "system:openshift:discovery"
```
Comment (enj) [inline pkg/cmd/server/bootstrappolicy/constants.go]: Drop the duplicate rules from this cluster role?

**Comment by enj** [comment]
Also does the `DiscoveryRoleName + "-binding"` need to go into `dead.go`?

File: pkg/cmd/server/bootstrappolicy/constants.go (hunk 2 above)
Comment (deads2k) [reply pkg/cmd/server/bootstrappolicy/constants.go]: > Drop the duplicate rules from this cluster role?

That would affect the scope authorizer amongst other spots.  This solves 90% of our problem.  I'd live with extra rules we can transition at some later point.

**Comment by deads2k** [comment]
> Which annotations is that?

system only and descriptions.

**Comment by deads2k** [comment]
> Also does the DiscoveryRoleName + "-binding" need to go into dead.go?

Only if we're worried about re-use I think.  Are we?

**Pull request merged**